docs/source/user_guide/user_guide/synchronization.md

```diff
@@ -0,0 +1,78 @@
+# Synchronization
+
+Since running code on GPUs is inherently multi-threaded, synchronization of writes and reads to memory is important.
+
+# Why is this an issue?
+
+- when one thread writes to global memory, the write doesn't take place immediately. It takes a long time (at computer speeds) for the data to work its way across to global memory, which is not on the GPU die itself, but in other chips, albeit on the same GPU card.
+    - think of it like sending a snail mail letter, whilst you work on other things, and lead your life
+- after the store statement has executed, the kernel continues to execute other statements, whilst the data works its way to memory
+```
+@ti.kernel f1(a: ti.Template) -> None:
+     a[0] = 5
+     # .. execution continues here immediately
+```
+
+Now, if another thread tries to read a[0], it will not get the new written value, but some much older value.
+
+To prevent this issue, we need some kind of synchronization.
+
+Synchronization of writes and reads for global memory is hard, because it is so far away from the cores; and takes such a long time to write, and read.
+
+# Possible approaches
+
+There are a few options:
+
+## Each thread never reads memory written by other threads
+
+- this approach is the safest, and the fastest
+- not always possible of course
```

Comment:
Strange to use bullet points for a single sentence by as you wish.

Reply:
oh, I'd forgotten about this doc. overlaps with my revised `paralellziation` doc now 🤔